src/OpenTelemetry/OpenTelemetry.csproj

```
@@ -21,4 +21,8 @@
     <ProjectReference Include="$(RepoRoot)\src\OpenTelemetry.Api.ProviderBuilderExtensions\OpenTelemetry.Api.ProviderBuilderExtensions.csproj" />
   </ItemGroup>
 
+  <ItemGroup>
+    <Compile Update="Internal\UnconditionalSuppressMessageAttribute.cs" LinkBase="Common" />
```

```suggestion
    <Compile Remove="Internal\UnconditionalSuppressMessageAttribute.cs" Condition="'$(TargetFrameworkIdentifier)' == '.NETCoreApp'" />
```

Since this file is under this .csproj folder, it is always going to be included for compilation. However, we don't need it when targeting `net6.0`, so we REMOVE the file when targeting `.NETCoreApp`

---

Another approach would be to put this "Common" file in a separate "src\Common" folder outside of all the projects. Then this .csproj could look just like all the others:

```xml
<Compile Include="$(RepoRoot)\src\<CommonFolder>\UnconditionalSuppressMessageAttribute.cs" LinkBase ="Common" Condition="'$(TargetFrameworkIdentifier)' != '.NETCoreApp'" />
```

---

Hey @eerhardt, would you mind sharing the doc that defines `.NETCoreApp`?
I'm not sure whether I'm looking at the latest one:
https://github.com/dotnet/designs/blob/main/accepted/2020/net5/net5.md
Thank you.

---

You can read more about target frameworks here: https://learn.microsoft.com/en-us/dotnet/standard/frameworks

`$(TargetFrameworkIdentifier)` identifies which .NET "Target framework" (without version) you are targeting. Some examples are:

* `.NETFramework`
* `.NETStandard`
* `.NETCoreApp`

That design document you linked is pretty good at explaining it as well. https://github.com/dotnet/designs/blob/main/accepted/2020/net5/net5.md#mapping-to-properties is a good section.

`Condition="'$(TargetFrameworkIdentifier)' == '.NETCoreApp'"` means: "When building for any of the frameworks like `netcoreapp3.1`, `net6.0`, `net8.0`, etc., do this logic." That condition won't be true when building for `netstandardX.Y` or any "traditional" .NET Framework targets.